ocs_ci/helpers/e2e_helpers.py

```diff
@@ -392,3 +411,337 @@ def validate_mcg_object_expiration(
+    )
+
+    # Validate noobaa pods are up and running
+    pod_obj = OCP(
+        kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"]
+    )
+    noobaa_pods = get_noobaa_pods()
+    pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        resource_count=len(noobaa_pods),
+        selector=constants.NOOBAA_APP_LABEL,
+        timeout=900,
+    )
+
+    # Verify everything running fine
+    logger.info("Verifying all resources are Running and matches expected result")
+    self.sanity_helpers.health_check(tries=120)
+
+    # Since the rebuild changed the noobaa-admin secret, update
+    # the s3 credentials in mcg_object_session
+    mcg_obj_session.update_s3_creds()
+
+    # Verify default backingstore/bucketclass
+    default_bs = OCP(
+        kind=constants.BACKINGSTORE, namespace=config.ENV_DATA["cluster_namespace"]
+    ).get(resource_name=DEFAULT_NOOBAA_BACKINGSTORE)
+    default_bc = OCP(
+        kind=constants.BUCKETCLASS, namespace=config.ENV_DATA["cluster_namespace"]
+    ).get(resource_name=DEFAULT_NOOBAA_BUCKETCLASS)
+    assert (
+        default_bs["status"]["phase"]
+        == default_bc["status"]["phase"]
+        == constants.STATUS_READY
+    ), "Failed: Default bs/bc are not in ready state"
+
+    # Create OBCs
+    logger.info("Creating OBCs after noobaa rebuild")
+    bucket_factory_session(amount=3, interface="OC", verify_health=True)
+
+
+def validate_noobaa_db_backup_recovery_locally_system(
+    self,
+    bucket_factory,
+    noobaa_db_backup_and_recovery_locally,
+    warps3,
+    mcg_obj_session,
+):
+    """
+    Test to verify Backup and Restore for Multicloud Object Gateway database locally
+    Backup procedure:
+        * Create a test bucket and write some data
+        * Backup noobaa secrets to local folder OR store it in secret objects
+        * Backup the PostgreSQL database and save it to a local folder
+        * For testing, write new data to show a little data loss between backup and restore
+    Restore procedure:
+        * Stop MCG reconciliation
+        * Stop the NooBaa Service before restoring the NooBaa DB.
+          There will be no object service after this point
+        * Verify that all NooBaa components (except NooBaa DB) have 0 replicas
+        * Login to the NooBaa DB pod and cleanup potential database clients to nbcore
+        * Restore DB from a local folder
+        * Delete current noobaa secrets and restore them from a local folder OR secrets objects.
+        * Restore MCG reconciliation
+        * Start the NooBaa service
+        * Restart the NooBaa DB pod
+        * Check that the old data exists, but not s3://testloss/
+    Run multi client warp benchamrking to verify bug https://bugzilla.redhat.com/show_bug.cgi?id=2141035
+
+    """
+
+    # create a bucket for warp benchmarking
+    bucket_name = bucket_factory()[0].name
+
+    # Backup and restore noobaa db using fixture
+    noobaa_db_backup_and_recovery_locally()
```

Comment:
test will fail as it uses the default function scoped fixture `bucket_factory()` so you need to pass the parent function argument bucket_factory which is session scoped

Reply:
done.